File: Project Euler/1-9/problem1.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

sum = 0
n = 1000

for i in xrange(0, n, 3):
    sum += i
for i in xrange(0, n, 5):
    sum += i
for i in xrange(0, n, 15):
    sum -= i

print sum

File: Project Euler/1-9/ruby/problem1.rb
#!/usr/bin/env ruby
# encoding: utf-8

def calculate_sum_from_zero(num, step)
  x = num / step
  return (x * (x + 1) / 2) * step
end

puts calculate_sum_from_zero(999, 3) + calculate_sum_from_zero(999, 5) - calculate_sum_from_zero(999, 15)
